### PR TITLE
[ZEPPELIN-777] Math formula support 

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
This PR adds support for formatting math formula formatting in %html display system using MathJax library.

### What type of PR is it?
Feature

### Todos
* [x] - Format math formula with MathJax library

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-777

### How should this be tested?
try run following codes.
```
%md
When \\(a \\ne 0\\), there are two solutions to \\(ax^2 + bx + c = 0\\) and they are
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
```
Note MathJax works better with `markdown.parser.type` property set `pegdown`, in markdown interpreter. With default markdown4j parser, some formula is not well displayed. (for example, ax^2) I think this will not be a big problem because of we'll remove markdown4j https://github.com/apache/zeppelin/pull/1594

```
%sh echo -e "%html \$\$a = b\$\$"
```

```
%spark println("%html $$b = c$$")
```

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update? yes
* Is there breaking changes for older versions? no
* Does this needs documentation? yes

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
cc @1ambda @AhyoungRyu

File: zeppelin-web/src/index.html [hunk 1]
```diff
@@ -100,6 +100,20 @@
     <![endif]-->
     <!-- endbuild -->
 
+    <script type="text/x-mathjax-config">
```
Comment (anthonycorbacho) [inline zeppelin-web/src/index.html]: Do you think we can move this part in a angular way like directive or something similar?

File: docs/displaysystem/basicdisplaysystem.md [hunk 2]
```diff
@@ -39,6 +39,13 @@ With `%html` directive, Zeppelin treats your output as HTML
 
 <img src="/assets/themes/zeppelin/img/screenshots/display_html.png" />
 
+### Mathmetical expressions
+HTML display system automatically format mathmetical expression using [MathJax](https://www.mathjax.org/). You can use
```
Comment (AhyoungRyu) [inline docs/displaysystem/basicdisplaysystem.md]: format -> formats

File: docs/interpreter/markdown.md [hunk 3]
```diff
@@ -53,6 +53,12 @@ The following example demonstrates the basic usage of Markdown in a Zeppelin not
 
 <img src="../assets/themes/zeppelin/img/docs-img/markdown-example.png" width="70%" />
 
+## Mathmetical expression
+
+Markdown interpreter leverage %html display system internally. That means you can mix mathmetical expressions with markdown syntax. For more information, please see [Mathmetical Expression](../displaysystem/basicdisplaysystem.html#mathmetical-expressions) section.
```
Comment (AhyoungRyu) [inline docs/interpreter/markdown.md]: leverage -> leverages

File: docs/displaysystem/basicdisplaysystem.md [hunk 4]
```diff
@@ -39,6 +39,13 @@ With `%html` directive, Zeppelin treats your output as HTML
 
 <img src="/assets/themes/zeppelin/img/screenshots/display_html.png" />
 
+### Mathmetical expressions
```
Comment (AhyoungRyu) [inline docs/displaysystem/basicdisplaysystem.md]: You mean `mathematical`?  :)

File: zeppelin-web/src/index.html (hunk 1 above)
Comment (Leemoonsoo) [inline zeppelin-web/src/index.html]: This is how MathJax is configured. And there's no needs to use this in anywhere else again. So i think it's better to leave it as is.

**Comment by Leemoonsoo** [comment]
@AhyoungRyu @anthonycorbacho addressed your comments.

**Comment by AhyoungRyu** [comment]
Tested and it works well as expected 👍

**Comment by tae-jun** [comment]
So cooool! 👍

**Comment by minahlee** [comment]
It works with dev mode (`grunt serve`) but doesn't work if you build source with mvn.
Attaching screenshot:
screen shot 2016-11-07 at 11 46 33 pm

**Comment by rawkintrevo** [comment]
@Leemoonsoo awesome work! so happy to see this coming through!

I am running into the same problem as @minahlee Thoughts on why `mvn` build wouldn't make this come out right?

**Comment by Leemoonsoo** [comment]
@minahlee @rawkintrevo I have fixed the problem. Now packaged version works as well.

**Comment by rawkintrevo** [comment]
@Leemoonsoo still not working for me, but it could be on my end- I have a hell of a time with git

Did a fetch /hard reset within the last hour.  
```bash
$ git fetch leemoonsoo
$ git reset --hard leemoonsoo/zeppelin-777
$ mvn clean package -DskipTests
<build success>
$  bin/zeppelin-daemon.sh start
```

Cleared browser cache / history / app data.

screenshot from 2016-11-08 10-57-45

**Comment by Leemoonsoo** [comment]
@rawkintrevo I clean cloned this PR, build and tried with browser cache clean, but couldn't reproduce the problem. Could you see `zeppelin-web/dist/jax` directory after the build?

**Comment by rawkintrevo** [comment]
lgtm. +1 to merge

again, awesome add @Leemoonsoo

**Comment by Leemoonsoo** [comment]
Thanks @rawkintrevo and @minahlee for the review.

Merge to master if there's no further comment.

**Comment by giaosudau** [comment]
I build a distribution from source code but it's failed because of missing /jax /extesions/
Did you guys test this case?
How to fix it?
Thanks.

**Comment by tae-jun** [comment]
@giaosudau Hi! Thanks for sharing the issue :)

Could you please share your building command line and error log?

I don't know why, but I guess MathJax library wasn't downloaded from bower.

**Comment by giaosudau** [comment]
@tae-jun 
No error at building source code.
But the error happens when I deploy with ZEPPELIN_SERVER_CONTEXT_PATH is /zeppelin/ 

Log

**Comment by tae-jun** [comment]
@giaosudau Thanks for the prompt response!

So you mean it doesn't happen without ZEPPELIN_SERVER_CONTEXT_PATH?
I've never used that option so I might not be helpful ^^;

But could you please check `$ZEPPELIN_HOME/zeppelin-web/bower_components/MathJax/jax` directory exists? To ensure that bower downloaded dependencies well 😄

**Comment by giaosudau** [comment]
@tae-jun 
I checked there is a folder MathJax in the source I use to build but what I mean here is the URL calling jax lib was wrong. You can check the URL of jax in the image above.
=>>> It must be append /zeppelin/jax will be right and I already tested if I change url to this it works.
That mean configuration of this library was wrong and I already have the library included.

**Comment by tae-jun** [comment]
@giaosudau You clarified the issue! Would you like to create a [JIRA](https://issues.apache.org/jira/browse/ZEPPELIN) ticket for that? :-) Then other developers will see the issue. Since this PR is already merged, so it seems to be handled on another PR. Thanks.

**Comment by giaosudau** [comment]
@tae-jun https://issues.apache.org/jira/browse/ZEPPELIN-1723

**Comment by tae-jun** [comment]
@giaosudau Thanks! 👍 I appreciate it 😄

**Pull request closed without merging**